README.md
# Jinsy

**Requires at least:** WordPress 4.4  
**Tested up to:** WordPress 4.9.7
**Stable tag:** 1.0.1  
**Version:** 1.0.0  
**License:** GNU General Public License v2 or later  
**License URI:** http://www.gnu.org/licenses/gpl-2.0.html  
**Tags:** blog, custom-logo, e-commerce, post-formats, grid-layout, one-column, two-columns, custom-background, custom-colors, custom-header, custom-menu, featured-image-header, featured-images, flexible-header, full-width-template, sticky-post, theme-options, threaded-comments, translation-ready

## Description

Jinsy Magazine is a multipurpose WordPress theme built on Argon design. The theme is fully responsive and provides a ton of customization options that give you the flexibility you need for a unique brand. To keep pace with the latest technology, Jinsy Magazine is fully compatible with the popular Elementor and Gutenberg page builders, and will work seamlessly if you plan to use it for Ecommerce. The theme was created on a single-page layout and comes with elegant parallax scrolling. The blog page has a grid system and lets you set from 1 to 4 columns with articles, also multiple articles from the same category can be set as featured posts. Jinsy Magazine is an optimal choice for startups, small businesses, agencies, magazines, news sites, and online shops. Overall, a friendly and modern theme providing custom sections and typography, catchy animations, call-to-action forms, and a lightweight interface. Moreover, it was built with SEO in mind.

## Frequently Asked Questions

* How to install this child theme?
 
1.After downloading the zip file, go to Appearance > Themes and click on the Install Themes tab.  
2.Click on the Upload link.  
3.Upload the zip file that you downloaded and click Install Now.  
4.Click Activate to use the theme you just installed.  

## Copyright

Jinsy WordPress Theme is a child theme of Hestia WordPress Theme, Copyright 2018 Themeisle
Jinsy Wordpress Theme is distributed under the terms of the GNU GPL

Jinsy bundles the following third-party resources:

### Scripts
Bootstrap, http://getbootstrap.com/ Copyright 2011-2015 Twitter, Inc  
**License:** MIT License  
Source: http://www.opensource.org/licenses/MIT  

### Images

License: All mystock.photos images are licensed under the terms of the Creative Commons Zero, http://creativecommons.org/publicdomain/zero/1.0/  

**Images used:**



CHANGELOG.md

 ### v1.0.4 - 2018-12-13 
 **Changes:** 
 * Update .distignore
 
 ### v1.0.3 - 2018-09-28 
 **Changes:** 
 * Code review
 
 ### v1.0.2 - 2018-09-14 
 **Changes:** 
 * New screenshot
 
 ### v1.0.1 - 2018-09-14 
 **Changes:** 
 * Update Travis
 
 ### v1.0.0 - 2018-09-13 
 **Changes:** 
 * Fix deploy system
 


tests/test-jinsy.php
<?php
/**
 * Class TestJinsy
 *
 * @package jinsy
 */
/**
 * Sample test case.
 */
class TestJinsy extends WP_UnitTestCase {
	/**
	 * A single example test.
	 */
	function test_generic() {
		// Replace this with some actual testing code.
		$this->assertTrue( defined( 'JINSY_VERSION' ) );
	}
}


inc/blog-header.php
<?php
/**
 * Blog header layouts
 *
 * @package jinsy-magazine
 */

/**
 * Disable Blog Header
 *
 * @param WP_Customize_Manager $wp_customize Theme Customizer object.
 */
function jinsy_magazine_blog_header_layout_controls( $wp_customize ) {

	/* Blog header layout control */
	$wp_customize->add_setting(
		'jinsy_magazine_blog_header_layout',
		array(
			'default' => true,
		)
	);

	$wp_customize->add_control(
		new WP_Customize_Control(
			$wp_customize,
			'jinsy_magazine_blog_header_layout',
			array(
				'label'    => esc_html__( 'Disable Blog Header', 'jinsy-magazine' ),
				'section'  => 'hestia_blog_layout',
				'type'     => 'checkbox',
				'priority' => 27,
			)
		)
	);
}

if ( wp_get_theme()->Template === 'hestia' ) {
	add_action( 'customize_register', 'jinsy_magazine_blog_header_layout_controls', 101 );
}

/**
 * Check if blog header should bne hidden
 */
function jinsy_magazine_blog_header_hidden() {

	$disable_blog_header = get_theme_mod( 'jinsy_magazine_blog_header_layout', true );

	return $disable_blog_header;
}

/**
 * Add class on body tag for the third layout on blog-home and archive page
 */
function jinsy_magazine_header_layout_body_classes() {
	return 'classic-blog';
}

/**
 * Add classes on wrapper element for enabling the third layout option
 */
function jinsy_header_layout_body_classes() {
	return 'wrapper classic-blog';
}

/**
 * Disable header from blog and archive pages
 */
function jinsy_magazine_disable_blog_header() {

	if ( jinsy_magazine_blog_header_hidden() ) {

		/* Change header on archive pages */
		remove_all_actions( 'hestia_before_archive_content' );
		$header_layout = new Hestia_Header_Layout_Manager();
		add_action( 'hestia_before_archive_content', array( $header_layout, 'post_page_header' ) );

		/* Change header on blog page */
		remove_all_actions( 'hestia_before_index_content' );
	}
}
add_action( 'wp_enqueue_scripts', 'jinsy_magazine_disable_blog_header', 90 );


/**
 * Add compatibility with the third header layout when blog header is disabled
 * Filtering body and .wrapper classes to have the proper style on pages without header
 */
function jinsy_magazine_blog_without_header_style() {

	if ( jinsy_magazine_blog_header_hidden() ) {
		if ( is_home() || ( is_archive() && ! ( class_exists( 'WooCommerce' ) && is_product_category() ) ) ) {
			add_filter( 'hestia_header_layout', 'jinsy_magazine_header_layout_body_classes', 30 );
			add_filter( 'hestia_page_wrapper_class', 'jinsy_header_layout_body_classes' );
		}
	}
}
add_action( 'wp_enqueue_scripts', 'jinsy_magazine_blog_without_header_style', 100 );


inc/inline-style.php
<?php
/**
 * Inline style
 *
 * @package jinsy-magazine
 */

/**
 * Inline style when parent theme is Hestia
 */
function jinsy_magazine_inline_style() {

	$accent_color = get_theme_mod( 'accent_color', '#5e72e4' );

	$custom_css = '';

	if ( ! empty( $accent_color ) && function_exists( 'hestia_hex_rgba' ) ) {
		$custom_css .= '
			.hestia-scroll-to-top,
			.hestia-scroll-to-top:hover,
			.hestia-scroll-to-top:focus, 
			.hestia-scroll-to-top:active {
				background-color: ' . esc_html( $accent_color ) . ';
			}
		';
		$custom_css .= '
			.hestia-scroll-to-top {
				-webkit-box-shadow: 0 2px 2px 0 ' . hestia_hex_rgba( $accent_color, '0.14' ) . ',0 3px 1px -2px ' . hestia_hex_rgba( $accent_color, '0.2' ) . ',0 1px 5px 0 ' . hestia_hex_rgba( $accent_color, '0.12' ) . ';
		    box-shadow: 0 2px 2px 0 ' . hestia_hex_rgba( $accent_color, '0.14' ) . ',0 3px 1px -2px ' . hestia_hex_rgba( $accent_color, '0.2' ) . ',0 1px 5px 0 ' . hestia_hex_rgba( $accent_color, '0.12' ) . ';
			}
		';
		$custom_css .= '
			.hestia-scroll-to-top:hover {
				-webkit-box-shadow: 0 14px 26px -12px' . hestia_hex_rgba( $accent_color, '0.42' ) . ',0 4px 23px 0 rgba(0,0,0,0.12),0 8px 10px -5px ' . hestia_hex_rgba( $accent_color, '0.2' ) . ';
		    box-shadow: 0 14px 26px -12px ' . hestia_hex_rgba( $accent_color, '0.42' ) . ',0 4px 23px 0 rgba(0,0,0,0.12),0 8px 10px -5px ' . hestia_hex_rgba( $accent_color, '0.2' ) . ';
			}
		';
	}

	if ( ! empty( $accent_color ) ) {
		$custom_css .= '
			.hestia-top-bar {
				background-color: ' . esc_html( $accent_color ) . ';
			}
		';
	}

	/* Hide Blog page title when Magazine Layout is enabled */
	$magazine_layout_enabled = get_theme_mod( 'jinsy_magazine_magazine_layout', true );
	if ( $magazine_layout_enabled ) {
		$custom_css .= '
			.blog .title-in-content {
				display: none;
			}
		';
	} else {
		$custom_css .= '
			.blog .title-in-content {
				display: block;
			}
		';
	}

	wp_add_inline_style( 'hestia_style', $custom_css );
}

if ( wp_get_theme()->Template === 'hestia' ) {
	add_action( 'wp_enqueue_scripts', 'jinsy_magazine_inline_style', 20 );
}


inc/disable-header.php
<?php
/**
 * Disable Header controls and functions
 *
 * @package jinsy-magazine
 */

/**
 * Disable Header customizer controls
 *
 * @param WP_Customize_Manager $wp_customize Theme Customizer object.
 */
function jinsy_magazine_disable_header_controls( $wp_customize ) {
	$wp_customize->add_setting(
		'jinsy_magazine_disable_header_layout',
		array(
			'default' => true,
		)
	);
	$wp_customize->add_control(
		new WP_Customize_Control(
			$wp_customize,
			'jinsy_magazine_disable_header_layout',
			array(
				'label'    => esc_html__( 'Disable Header', 'jinsy-magazine' ),
				'section'  => 'header_image',
				'type'     => 'checkbox',
				'priority' => 10,
			)
		)
	);

	$object = $wp_customize->get_control( 'hestia_header_layout' );

	if ( ! empty( $object ) ) {
		$object->active_callback = 'jinsy_magazine_disable_header_active_callback';
	}
}
add_action( 'customize_register', 'jinsy_magazine_disable_header_controls', 101 );

/**
 * Hide header layout controls if header is disabled
 */
function jinsy_magazine_disable_header_active_callback() {
	if ( get_theme_mod( 'jinsy_magazine_disable_header_layout', true ) === true ) {
		return false;
	}
	return true;
}

/**
 * Third option of header layout control
 */
function jinsy_magazine_third_header_layout() {
	return 'classic-blog';
}

/**
 * Filter header layout in order to disable the header
 */
function jinsy_magazine_disable_header() {

	$disable_header = get_theme_mod( 'jinsy_magazine_disable_header_layout', true );

	if ( $disable_header ) {
		add_filter( 'theme_mod_hestia_header_layout', 'jinsy_magazine_third_header_layout' );
		remove_all_actions( 'hestia_before_index_content' );
	} else {
		remove_filter( 'theme_mod_hestia_header_layout', 'jinsy_magazine_third_header_layout' );
	}
}
add_action( 'wp', 'jinsy_magazine_disable_header', 0 );


inc/magazine-layout.php
<?php
/**
 * Functions for magazine layout
 *
 * @package jinsy-magazine
 */

/**
 * Customizer controls for magazine layout feature
 *
 * @param WP_Customize_Manager $wp_customize Theme Customizer object.
 *
 * @since 1.0.0
 */
function jinsy_magazine_customize_register( $wp_customize ) {

	/* Magazine layout control */
	$wp_customize->add_setting(
		'jinsy_magazine_magazine_layout',
		array(
			'default' => true,
		)
	);

	$wp_customize->add_control(
		new WP_Customize_Control(
			$wp_customize,
			'jinsy_magazine_magazine_layout',
			array(
				'label'    => esc_html__( 'Enable Magazine Layout', 'jinsy-magazine' ),
				'section'  => 'hestia_blog_layout',
				'type'     => 'checkbox',
				'priority' => 26,
			)
		)
	);

	if ( jinsy_magazine_pro() ) {
		$magazine_layout_control                  = $wp_customize->get_control( 'jinsy_magazine_magazine_layout' );
		$magazine_layout_control->label           = esc_html__( 'Hide Blog Title', 'jinsy-magazine' );
		$magazine_layout_control->active_callback = 'jinsy_magazine_hide_blog_title_active_callback';

	} else {
		$object = $wp_customize->get_control( 'hestia_alternative_blog_layout' );

		if ( ! empty( $object ) ) {
			$object->active_callback = 'jinsy_magazine_magazine_layout_active_callback';
		}

		$object = $wp_customize->get_control( 'hestia_pagination_type' );

		if ( ! empty( $object ) ) {
			$object->active_callback = 'jinsy_magazine_magazine_layout_active_callback';
		}
	}
}
add_action( 'customize_register', 'jinsy_magazine_customize_register', 100 );

/**
 * Active callback for hide blog title control
 */
function jinsy_magazine_hide_blog_title_active_callback() {

	$static_blog_page_id = get_option( 'page_for_posts' );
	if ( ! empty( $static_blog_page_id ) ) {
		$static_blog_page_meta = get_post_meta( $static_blog_page_id );
	}
	$header_layout_control = get_theme_mod( 'hestia_header_layout', 'default' );

	if ( ( ! empty( $static_blog_page_meta['hestia_header_layout'] ) ) && ( $static_blog_page_meta['hestia_header_layout'][0] !== 'default' ) ) {
		return true;
	}

	if ( $header_layout_control !== 'default' ) {
		return true;
	}

	return false;
}

/**
 * Hide blog layout controls when magazine layout is enabled
 *
 * @return bool
 */
function jinsy_magazine_magazine_layout_active_callback() {
	if ( get_theme_mod( 'jinsy_magazine_magazine_layout', true ) === true ) {
		return false;
	}
	return true;
}

/**
 * Blog layout when magazine layout is enabled
 *
 * @return string
 */
function jinsy_magazine_grid_layout() {
	return 'blog_alternative_layout2';
}

/**
 * Number of columns for the blog grid when magazine layout is enabled
 *
 * @return int
 */
function jinsy_magazine_grid_layout_columns() {
	return 3;
}

/**
 * Enable masonry when magazine layout is enabled
 *
 * @return bool
 */
function jinsy_magazine_enable_masonry() {
	return true;
}

/**
 * Disable infinite scroll when magazine layout is enabled
 *
 * @return string
 */
function jinsy_magazine_pagination_type() {
	return 'number';
}

/**
 * Set classic blog for header layout while magazine layout is enabled
 */
function jinsy_magazine_classic_blog_header_layout() {
	return 'classic-blog';
}

/**
 * Magazine layout
 */
function jinsy_magazine_enable_magazine_layout() {

	$enable_magazine_layout = get_theme_mod( 'jinsy_magazine_magazine_layout', true );

	if ( $enable_magazine_layout && ! jinsy_magazine_pro() ) {
		add_filter( 'theme_mod_hestia_alternative_blog_layout', 'jinsy_magazine_grid_layout' );
		add_filter( 'theme_mod_hestia_grid_layout', 'jinsy_magazine_grid_layout_columns' );
		add_filter( 'theme_mod_hestia_enable_masonry', 'jinsy_magazine_enable_masonry' );
		add_filter( 'theme_mod_hestia_pagination_type', 'jinsy_magazine_pagination_type' );
		add_filter( 'hestia_header_layout', 'jinsy_magazine_classic_blog_header_layout' );
		remove_all_actions( 'hestia_before_index_content' );
	}
}
add_action( 'wp', 'jinsy_magazine_enable_magazine_layout', 0 );


inc/customizer/customizer.php
<?php
/**
 * Customizer functionality for the theme
 *
 * @package jinsy
 * @since 1.0.0
 */

function jinsy_customize_register( $wp_customize ) {

	/* Jinsy Options Panel */
	$wp_customize->add_panel(
		'jinsy_options',
		array(
			'priority' => 50,
			'title'    => esc_html__( 'Jinsy Options', 'jinsy' ),
		)
	);

	/* Features section */
	$wp_customize->add_section(
		'jinsy_features_settings',
		array(
			'priority' => 5,
			'title'    => esc_html__( 'Features Section', 'jinsy' ),
			'panel'    => 'jinsy_options',
		)
	);
	/* Features color control */
	$wp_customize->add_setting(
		'jinsy_features_background_color',
		array(
			'default' => '#ffffff',
		)
	);
	$wp_customize->add_control(
		new WP_Customize_Color_Control(
			$wp_customize,
			'jinsy_features_background_color',
			array(
				'priority' => 5,
				'label'    => 'Features Background Color',
				'section'  => 'jinsy_features_settings',
			)
		)
	);

	/* Shop section */
	$wp_customize->add_section(
		'jinsy_shop_settings',
		array(
			'priority' => 10,
			'title'    => esc_html__( 'Shop Section ', 'jinsy' ),
			'panel'    => 'jinsy_options',
		)
	);
	/* Shop color control */
	$wp_customize->add_setting(
		'jinsy_shop_background_color',
		array(
			'default' => '#ffffff',
		)
	);
	$wp_customize->add_control(
		new WP_Customize_Color_Control(
			$wp_customize,
			'jinsy_shop_background_color',
			array(
				'priority' => 5,
				'label'    => 'Shop Background Color',
				'section'  => 'jinsy_shop_settings',
			)
		)
	);

	/* Portfolio section */
	$wp_customize->add_section(
		'jinsy_portfolio_settings',
		array(
			'priority' => 15,
			'title'    => esc_html__( 'Portfolio Section ', 'jinsy' ),
			'panel'    => 'jinsy_options',
		)
	);
	/* Portfolio color control */
	$wp_customize->add_setting(
		'jinsy_portfolio_background_color',
		array(
			'default' => '#ffffff',
		)
	);
	$wp_customize->add_control(
		new WP_Customize_Color_Control(
			$wp_customize,
			'jinsy_portfolio_background_color',
			array(
				'priority' => 5,
				'label'    => 'Portfolio Background Color',
				'section'  => 'jinsy_portfolio_settings',
			)
		)
	);

	/* Team section */
	$wp_customize->add_section(
		'jinsy_team_settings',
		array(
			'priority' => 20,
			'title'    => esc_html__( 'Team Section ', 'jinsy' ),
			'panel'    => 'jinsy_options',
		)
	);
	/* Team color control */
	$wp_customize->add_setting(
		'jinsy_team_background_color',
		array(
			'default' => '#ffffff',
		)
	);
	$wp_customize->add_control(
		new WP_Customize_Color_Control(
			$wp_customize,
			'jinsy_team_background_color',
			array(
				'priority' => 5,
				'label'    => 'Team Background Color',
				'section'  => 'jinsy_team_settings',
			)
		)
	);

	/* Pricing section */
	$wp_customize->add_section(
		'jinsy_pricing_settings',
		array(
			'priority' => 25,
			'title'    => esc_html__( 'Pricing Section ', 'jinsy' ),
			'panel'    => 'jinsy_options',
		)
	);
	/* Pricing color control */
	$wp_customize->add_setting(
		'jinsy_pricing_background_color',
		array(
			'default' => '#ffffff',
		)
	);
	$wp_customize->add_control(
		new WP_Customize_Color_Control(
			$wp_customize,
			'jinsy_pricing_background_color',
			array(
				'priority' => 5,
				'label'    => 'Pricing Background Color',
				'section'  => 'jinsy_pricing_settings',
			)
		)
	);

	/* Testimonials section */
	$wp_customize->add_section(
		'jinsy_testimonials_settings',
		array(
			'priority' => 30,
			'title'    => esc_html__( 'Testimonials Section ', 'jinsy' ),
			'panel'    => 'jinsy_options',
		)
	);
	/* Testimonials color control */
	$wp_customize->add_setting(
		'jinsy_testimonials_background_color',
		array(
			'default' => '#ffffff',
		)
	);
	$wp_customize->add_control(
		new WP_Customize_Color_Control(
			$wp_customize,
			'jinsy_testimonials_background_color',
			array(
				'priority' => 5,
				'label'    => 'Testimonials Background Color',
				'section'  => 'jinsy_testimonials_settings',
			)
		)
	);

	/* Clients Bar section */
	$wp_customize->add_section(
		'jinsy_clients_bar_settings',
		array(
			'priority' => 35,
			'title'    => esc_html__( 'Clients Bar Section ', 'jinsy' ),
			'panel'    => 'jinsy_options',
		)
	);
	/* Clients Bar color control */
	$wp_customize->add_setting(
		'jinsy_clients_bar_background_color',
		array(
			'default' => '#ffffff',
		)
	);
	$wp_customize->add_control(
		new WP_Customize_Color_Control(
			$wp_customize,
			'jinsy_clients_bar_background_color',
			array(
				'priority' => 5,
				'label'    => 'Clients Bar Background Color',
				'section'  => 'jinsy_clients_bar_settings',
			)
		)
	);

	/* Blog section */
	$wp_customize->add_section(
		'jinsy_blog_settings',
		array(
			'priority' => 40,
			'title'    => esc_html__( 'Blog Section ', 'jinsy' ),
			'panel'    => 'jinsy_options',
		)
	);
	/* Blog color control */
	$wp_customize->add_setting(
		'jinsy_blog_background_color',
		array(
			'default' => '#ffffff',
		)
	);
	$wp_customize->add_control(
		new WP_Customize_Color_Control(
			$wp_customize,
			'jinsy_blog_background_color',
			array(
				'priority' => 5,
				'label'    => 'Blog Background Color',
				'section'  => 'jinsy_blog_settings',
			)
		)
	);

}

add_action( 'customize_register', 'jinsy_customize_register', 99 );


functions.php
<?php
/**
 * Jinsy Magazine functions and definitions.
 *
 * @package jinsy-magazine
 * @since 1.0.0
 */

define( 'JINSY_MAGAZINE_VERSION', '1.0.4' );

$vendor_file = trailingslashit( get_stylesheet_directory() ) . 'vendor/autoload.php';
if ( is_readable( $vendor_file ) ) {
	require_once $vendor_file;
}
add_filter(
	'themeisle_sdk_products',
	function ( $products ) {
		$products[] = get_stylesheet_directory() . '/style.css';
		return $products;
	}
);

if ( ! defined( 'ABSPATH' ) ) {
	exit;
}

if ( ! function_exists( 'jinsy_magazine_parent_css' ) ) :
	/**
	 * Enqueue parent style
	 *
	 * @since 1.0.0
	 */
	function jinsy_magazine_parent_css() {
		wp_enqueue_style( 'jinsy_magazine_parent', trailingslashit( get_template_directory_uri() ) . 'style.css', array( 'bootstrap' ), JINSY_MAGAZINE_VERSION );
		wp_style_add_data( 'jinsy_magazine_parent', 'rtl', 'replace' );
		wp_style_add_data( 'hestia_style', 'rtl', 'replace' );
	}

endif;
add_action( 'wp_enqueue_scripts', 'jinsy_magazine_parent_css', 10 );

/**
 * Check if parent theme is Hestia Pro
 */
function jinsy_magazine_pro() {
	if ( wp_get_theme()->Template === 'hestia-pro' ) {
		return true;
	}
	return false;
}

if ( file_exists( get_stylesheet_directory() . '/inc/magazine-layout.php' ) ) {
	require_once get_stylesheet_directory() . '/inc/magazine-layout.php';
}

if ( file_exists( get_stylesheet_directory() . '/inc/inline-style.php' ) ) {
	require_once get_stylesheet_directory() . '/inc/inline-style.php';
}

/**
 * Default accent color
 */
function jinsy_magazine_accent_color_default() {
	return '#5e72e4';
}

add_filter( 'hestia_accent_color_default', 'jinsy_magazine_accent_color_default' );

/**
 * Default header gradient color
 */
function jinsy_magazine_header_gradient_default() {
	return '#8965e0';
}

add_filter( 'hestia_header_gradient_default', 'jinsy_magazine_header_gradient_default' );

/**
 * Default font
 */
function jinsy_magazine_font_default() {
	return 'Open Sans';
}

add_filter( 'hestia_headings_default', 'jinsy_magazine_font_default' );
add_filter( 'hestia_body_font_default', 'jinsy_magazine_font_default' );

/**
 * Default font in customizer preview
 *
 * @param object $wp_customize the customizer manager.
 */
function jinsy_magazine_customizer_font_default( $wp_customize ) {

	$jinsy_magazine_headings_font = $wp_customize->get_setting( 'hestia_headings_font' );
	if ( ! empty( $jinsy_magazine_headings_font ) ) {
		$jinsy_magazine_headings_font->default = jinsy_magazine_font_default();
	}

	$jinsy_magazine_body_font = $wp_customize->get_setting( 'hestia_body_font' );
	if ( ! empty( $jinsy_magazine_body_font ) ) {
		$jinsy_magazine_body_font->default = jinsy_magazine_font_default();
	}
}
add_action( 'customize_register', 'jinsy_magazine_customizer_font_default', 99 );

/**
 * Default boxed-layout
 */
function jinsy_magazine_boxed_layout_default() {
	return 0;
}

add_filter( 'hestia_boxed_layout_default', 'jinsy_magazine_boxed_layout_default' );

/**
 * Default header alignment - center
 */
function jinsy_magazine_header_alignemnt_default() {
	return 'center';
}
add_filter( 'hestia_header_alignment_default', 'jinsy_magazine_header_alignemnt_default' );

/**
 * Default header layout - classic blog, featured image under the title
 */
function jinsy_magazine_header_layout_default() {
	return 'classic-blog';
}
add_filter( 'hestia_header_layout_default', 'jinsy_magazine_header_layout_default' );

/**
 * Default padding for buttons styling
 */
function jinsy_magazine_button_padding_dimension_default() {
	return json_encode(
		array(
			'desktop' => json_encode(
				array(
					'desktop_vertical'   => 10,
					'desktop_horizontal' => 20,
				)
			),
		)
	);
}

add_filter( 'hestia_button_padding_dimensions_default', 'jinsy_magazine_button_padding_dimension_default' );

/**
 * Default border-radius for buttons styling
 */
function jinsy_magazine_buttons_border_radius_default() {
	return 4;
}

add_filter( 'hestia_buttons_border_radius_default', 'jinsy_magazine_buttons_border_radius_default' );

/**
 * Default background type for Big Title section
 */
function jinsy_magazine_big_title_background_default() {
	return 'parallax';
}

add_filter( 'hestia_slider_type_default', 'jinsy_magazine_big_title_background_default' );

/**
 * Default image for parallax layer1 in Big Title section background
 *
 * @return string
 */
function jinsy_magazine_parallax_layer1_default() {
	return get_stylesheet_directory_uri() . '/assets/img/parallax_layer1.jpg';
}

add_action( 'hestia_parallax_layer1_default', 'jinsy_magazine_parallax_layer1_default' );

/**
 * Default image for parallax layer2 in Big Title section background
 *
 * @return string
 */
function jinsy_magazine_parallax_layer2_default() {
	return get_stylesheet_directory_uri() . '/assets/img/parallax_layer2.png';
}

add_action( 'hestia_parallax_layer2_default', 'jinsy_magazine_parallax_layer2_default' );

/**
 * Default color for header overlay
 *
 * @return string
 */
function jinsy_magazine_header_overlay_color_default() {
	return 'rgba(0,0,0,0.2)';
}
add_action( 'header_overlay_color', 'jinsy_magazine_header_overlay_color_default' );

/**
 * Enable scroll to top by default
 */
function jinsy_magazine_enable_scroll_to_top() {
	return 1;
}
add_filter( 'hestia_scroll_to_top_default', 'jinsy_magazine_enable_scroll_to_top' );

/**
 * Import options from Hestia
 *
 * @since 1.0.0
 */
function jinsy_magazine_import_hestia_options() {

	$previous_theme = strtolower( get_option( 'theme_switched' ) );
	$child_template = wp_get_theme()->Template;

	if ( $previous_theme !== $child_template ) {
		return;
	}

	$allowed_themes = array( 'hestia', 'hestia-pro' );
	if ( ! in_array( $previous_theme, $allowed_themes ) ) {
		return;
	}

	$hestia_mods = get_option( 'theme_mods_' . $child_template );
	if ( empty( $hestia_mods ) ) {
		return;
	}

	foreach ( $hestia_mods as $hestia_mod_k => $hestia_mod_v ) {
		set_theme_mod( $hestia_mod_k, $hestia_mod_v );
	}

}

add_action( 'after_switch_theme', 'jinsy_magazine_import_hestia_options' );

/**
 * Change default welcome notice that appears after theme first installed
 */
function jinsy_magazine_welcome_notice_filter() {

	$theme = wp_get_theme();

	$theme_name = $theme->get( 'Name' );
	$theme      = $theme->parent();

	$theme_slug = $theme->get_template();

	$var = '<p>' . sprintf( 'Welcome! Thank you for choosing %1$s! To fully take advantage of the best our theme can offer please make sure you visit our %2$swelcome page%3$s.', $theme_name, '<a href="' . esc_url( admin_url( 'themes.php?page=' . $theme_slug . '-welcome' ) ) . '">', '</a>' ) . '</p><p><a href="' . esc_url( admin_url( 'themes.php?page=' . $theme_slug . '-welcome' ) ) . '" class="button" style="text-decoration: none;">' . sprintf( 'Get started with %s', $theme_name ) . '</a></p>';

	return wp_kses_post( $var );
}

add_filter( 'hestia_welcome_notice_filter', 'jinsy_magazine_welcome_notice_filter' );

/**
 * Change About page defaults
 *
 * @param string $old_value Old value beeing filtered.
 * @param string $parameter Specific parameter for filtering.
 */
function jinsy_magazine_about_page_filter( $old_value, $parameter ) {

	switch ( $parameter ) {
		case 'menu_name':
		case 'pro_menu_name':
			$return = esc_html__( 'About Jinsy Magazine', 'jinsy-magazine' );
			break;
		case 'page_name':
		case 'pro_page_name':
			$return = esc_html__( 'About Jinsy Magazine', 'jinsy-magazine' );
			break;
		case 'welcome_title':
		case 'pro_welcome_title':
			/* translators: s - theme name */
			$return = sprintf( esc_html__( 'Welcome to %s! - Version ', 'jinsy-magazine' ), 'Jinsy Magazine' );
			break;
		case 'welcome_content':
		case 'pro_welcome_content':
			$return = esc_html__( 'Jinsy Magazine is a multipurpose WordPress theme built on Argon design. The theme is fully responsive and provides a ton of customization options that give you the flexibility you need for a unique brand. To keep pace with the latest technology, Jinsy Magazine is fully compatible with the popular Elementor and Gutenberg page builders, and will work seamlessly if you plan to use it for Ecommerce. The theme was created on a single-page layout and comes with elegant parallax scrolling. The blog page has a grid system and lets you set from 1 to 4 columns with articles, also multiple articles from the same category can be set as featured posts. Jinsy Magazine is an optimal choice for startups, small businesses, agencies, magazines, news sites, and online shops. Overall, a friendly and modern theme providing custom sections and typography, catchy animations, call-to-action forms, and a lightweight interface. Moreover, it was built with SEO in mind.', 'jinsy-magazine' );
			break;
		default:
			$return = '';
	}

	return $return;
}

add_filter( 'hestia_about_page_filter', 'jinsy_magazine_about_page_filter', 0, 3 );

/**
 * Declare text domain for this child theme.
 * Translations can be filed in the /languages/ directory.
 */
function jinsy_magazine_theme_setup() {
	load_child_theme_textdomain( 'jinsy-magazine', get_stylesheet_directory() . '/languages' );
}

add_action( 'after_setup_theme', 'jinsy_magazine_theme_setup' );

